### ISSUE TITLE
Singleplayer button on the main menu isn't working with 0.11.12.00 installed.

### ISSUE THREAD

**Issue by belathus** [XCompWiz/Mystcraft-Issues#98]
I'm updating the Wanderlust Reloaded modpack. It has CustomMainMenu, and I'm using Forge build 1448. Everything works fine with 0.11.11.00, but updating to 0.11.12.00, the single player button acts like I'm using the 'reload' button, rather than starting the SinglePlayer GUI. Nothing is displayed on the log when the button is pressed.

**Comment by Wolf68k** [comment]
I'm having the same problem with DNSTechPack. When I disabled CMM and restart the pack it throws me into a creative world. Skips the main menu, loads a creative world and begins a ever so slow loading bar about doing first-time profiling and that I must wait for it to finish. With CMM enabled I'm not sure if this is still happening in the background or what.

**Comment by XCompWiz** [comment]
Servers didn't change and still do the profiling in the background. They also still have the config option to prevent player connections while profiling is running.

This is clearly a conflict between the improved Mystcraft profiling system and the CustomMainMenu mod.  Given what you describe, I can conclude that the custom main menu mod isn't extending the vanilla interfaces, meaning Mystcraft cannot detect if we are on the main menu or not.  Chances are the profiling doesn't happen, as Mystcraft waits for the main menu before it does it's error checking, and the profiling happens after that.  I might be able to rearrange things, but I'd really appreciate if someone could point the dev of the CustomMainMenu mod to this issue so we could determine the best course of action.

**Comment by Wolf68k** [comment]
After disabling CMM and letting Mystcraft do it's thing. I reloaded the pack with CMM enabled and still the Single Player button doesn't want to work.

**Comment by XCompWiz** [comment]
The single player button working is contingent on the load up system finishing.  Since you never open a "MainMenu" object, the load up system never starts.

**Comment by Wolf68k** [comment]
I think that went a bit over my head, so I'll just nod and agree and act like I know what you said.

Also I reported it to Lumien as well and pointed a link here as you suggested.

**Comment by lumien231** [comment]
Assuming that you use GuiOpenEvent to detect the opening of the main menu right now would it be possible for you to switch to InitGuiEvent.Post? I do populate that gui field with a fake main menu that extends GuiMainMenu.

**Comment by XCompWiz** [comment]
I can try that, sure. :)  Thanks for the speedy and useful help. :D

**Comment by XCompWiz** [comment]
OK, I've changed the initial startup to the following:
```Java
	@SubscribeEvent
	public void onClientInitGUI(InitGuiEvent.Post event) {
		if (event.gui instanceof GuiMainMenu) {
			if (checkForErrors()) MystcraftFirstRun.enable();
		}
	}
```

Will that play nicely with the custom main menus, lumien? :)

Note that enable doesn't actually start it. It starts on the following client tick (as some loadup stuff has to happen).
I'm still using `GuiOpenEvent` for some checks, but I don't think there will be any issues there.  The main issue here was that I was, in the client tick, checking if the currently open gui was the main menu; and it wasn't, obviously.  I had needed to wait on the first client tick anyway, so I just put the check there.
